src/main/java/com/arpnetworking/tsdcore/sinks/MetricsDataSerializer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.sinks;
+
+import com.arpnetworking.metrics.aggregation.protocol.Messages;
+import com.arpnetworking.metrics.mad.model.AggregatedData;
+import com.arpnetworking.metrics.mad.model.statistics.HistogramStatistic;
+import com.arpnetworking.metrics.mad.model.statistics.Statistic;
+import com.arpnetworking.metrics.mad.model.statistics.StatisticFactory;
+import com.arpnetworking.tsdcore.model.AggregationMessage;
+import com.arpnetworking.tsdcore.model.PeriodicData;
+import com.google.protobuf.ByteString;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Utility class to serialize to metrics-aggregator-protocol protobuf messages.
+ *
+ * @author William Ehlhardt (whale at dropbox dot com)
+ */
+public final class MetricsDataSerializer {
+    private static final StatisticFactory STATISTIC_FACTORY = new StatisticFactory();
+    private static final Statistic EXPRESSION_STATISTIC = STATISTIC_FACTORY.getStatistic("expression");
+
+    /**
+     * Serialize a metric's data to a StatisticSetRecord.
+     *
+     * @param periodicData Originating PeriodicData
```

Comment:
This is probably why this was not refactored into a separate class; the periodicData is not the correct abstraction to be passed in here. It's somewhat redundant as it actually contains the metric name and AggregatedData that are being serialized.

Absent, either taking the individual parts of PeriodicData (yuck) or refactoring the model and/or serialization (haven't thought about how, open to ideas if you prefer this), the solution is probably to:

* Make this an instantiated class which accepts `PeriodicData` in its constructor
* Store the parts of `PeriodicData` you need as members (e.g. period, start, dimensions).
* Pass metricName and `AggregatedData` only to `createStatisticSetRecord` (aka `serializeMetricData`)

Alternatively, the constructor could accept the parts of `PeriodicData` (e.g. period, start, dimensions) to make it a little cleaner.

===

The refactoring that probably actually needs to happen is separating the entirely serialization routine from the sinks and including `AggregationMessage` and have a class which accepts a `PeriodicData` and spits out a byte[] (or some PB equivalent).

However, this is a bit academic since we're moving away from the Tcp based aggregation sink and it's being removed in MAD v2.